etl/steps/__init__.py

```diff
@@ -666,13 +665,12 @@ def _cached_dvc_repo(self) -> Any:
     def run(self) -> None:
         from dvc.exceptions import CheckoutError
 
-        with _unignore_backports(Path(self._path)):
+        with dvc_pull_lock, _unignore_backports(Path(self._path)):
             try:
                 repo = self._cached_dvc_repo()
```

Comment:
So, this PR is basically about putting this line inside the lock. Can you explain what that does? (both here, and in a comment?)